backend/src/plugins/mission/mission_service_impl.h

```diff
@@ -83,23 +108,92 @@ class MissionServiceImpl final : public dronecore::rpc::mission::MissionService:
+                    auto rpc_mission_result = generateRPCMissionResult(result);
+                    response->set_allocated_mission_result(rpc_mission_result);
+                }
+
+                result_promise.set_value();
+            });
+
+        result_future.wait();
+        return grpc::Status::OK;
+    }
+
+    grpc::Status GetCurrentMissionItemIndex(
+        grpc::ServerContext * /* context */,
+        const rpc::mission::GetCurrentMissionItemIndexRequest * /* request */,
+        rpc::mission::GetCurrentMissionItemIndexResponse *response) override
+    {
+        if (response != nullptr) {
+            response->set_index(_mission.current_mission_item());
         }
 
-        return mission_items;
+        return grpc::Status::OK;
+    }
+
+    grpc::Status GetMissionCount(grpc::ServerContext * /* context */,
+                                 const rpc::mission::GetMissionCountRequest * /* request */,
+                                 rpc::mission::GetMissionCountResponse *response) override
+    {
+        if (response != nullptr) {
+            response->set_count(_mission.total_mission_items());
+        }
+
+        return grpc::Status::OK;
+    }
+
+    static void translateMissionItem(const std::shared_ptr<MissionItem> mission_item,
+                                     rpc::mission::MissionItem *rpc_mission_item)
+    {
+        rpc_mission_item->set_latitude_deg(mission_item->get_latitude_deg());
+        rpc_mission_item->set_longitude_deg(mission_item->get_longitude_deg());
+        rpc_mission_item->set_relative_altitude_m(mission_item->get_relative_altitude_m());
+        rpc_mission_item->set_speed_m_s(mission_item->get_speed_m_s());
+        rpc_mission_item->set_is_fly_through(mission_item->get_fly_through());
+        rpc_mission_item->set_gimbal_pitch_deg(mission_item->get_gimbal_pitch_deg());
+        rpc_mission_item->set_gimbal_yaw_deg(mission_item->get_gimbal_yaw_deg());
+        rpc_mission_item->set_camera_action(
+            translateCameraAction(mission_item->get_camera_action()));
+    }
+
+    static rpc::mission::MissionItem::CameraAction
+    translateCameraAction(const MissionItem::CameraAction camera_action)
+    {
+        switch (camera_action) {
+            case MissionItem::CameraAction::TAKE_PHOTO:
+                return rpc::mission::MissionItem_CameraAction_TAKE_PHOTO;
+            case MissionItem::CameraAction::START_PHOTO_INTERVAL:
+                return rpc::mission::MissionItem_CameraAction_START_PHOTO_INTERVAL;
+            case MissionItem::CameraAction::STOP_PHOTO_INTERVAL:
+                return rpc::mission::MissionItem_CameraAction_STOP_PHOTO_INTERVAL;
+            case MissionItem::CameraAction::START_VIDEO:
+                return rpc::mission::MissionItem_CameraAction_START_VIDEO;
+            case MissionItem::CameraAction::STOP_VIDEO:
+                return rpc::mission::MissionItem_CameraAction_STOP_VIDEO;
+            case MissionItem::CameraAction::NONE:
```

Comment:
I thought we need to add the comment `// FALLTHROUGH` for latest GCC but given CI doesn't complain it seems to be fine.

Reply:
Oh, I did not know about that. Good to know!